### ISSUE TITLE
Enumerating channel database files includes those that are broken symlinks

### ISSUE THREAD

**Issue by jamalex** [learningequality/kolibri#5370]
### Observed behavior

On a RACHEL, we create a symlink within the `.kolibri/content/databases/` directory to the content databases in the RACHEL modules directory. Uninstalling a module is meant to remove the symlinks, but if the module files are somehow deleted without the cleanup scripts being run, the symlinks may be left behind but be broken.

When we then enumerate the channel database files on upgrade, it includes those that are broken symlinks, but then when we try to actually load them, it fails to load the file.

```
Traceback (most recent call last):
  File "/usr/bin/kolibri", line 9, in <module>
    load_entry_point('kolibri==0.12.2', 'console_scripts', 'kolibri')()
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 673, in main
    start(port, daemon=daemon)
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 281, in start
    update()
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 260, in update
    update_channel_metadata()
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/annotation.py", line 44, in update_channel_metadata
    channel_ids = get_channel_ids_for_content_database_dir(get_content_database_dir_path())
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/channels.py", line 38, in get_channel_ids_for_content_database_dir
    if os.path.getsize(filename) == 0:
  File "/usr/lib/python3.4/genericpath.py", line 50, in getsize
    return os.stat(filename).st_size
FileNotFoundError: [Errno 2] No such file or directory: '/root/.kolibri/content/databases/197934f144305350b5820c7c4dd8e194.sqlite3'

root@WAPD-235N-Server:~# ls /root/.kolibri/content/databases/
197934f144305350b5820c7c4dd8e194.sqlite3  4da5cdd8cb7d431f9666c50ebcf9290f.sqlite3  a9b25ac9814742c883ce1b0579448337.sqlite3
1ceff53605e55bef987d88e0908658c5.sqlite3  a1e879bdbe7647d3aa73e8e3e753e1dd.sqlite3  e409b964366a59219c148f2aaa741f43.sqlite3
```

### Expected behavior

When we enumerate the files in the databases directory, we should also do an `exists` or `isfile` or whatnot on them to ensure they'll be readable. And/or handle exceptions reading a database and just skipping it.

### User-facing consequences

Led to a non-functional RACHEL on upgrade, as it didn't get past the error above.

### Steps to reproduce

Put a broken symlink at `.kolibri/content/databases/197934f144305350b5820c7c4dd8e194.sqlite3`, then delete `.kolibri/.data_version` and start Kolibri.

### Context

Upgrading to 0.12.2 on a RACHEL+

**Comment by jamalex** [comment]
A similar issue occurred on the same device with a corrupted (incompletely downloaded) content database. We should handle and skip in this case as well:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 1193, in _execute_context
    context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/default.py", line 509, in do_execute
    cursor.execute(statement, parameters)
sqlite3.DatabaseError: database disk image is malformed

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/bin/kolibri", line 9, in <module>
    load_entry_point('kolibri==0.12.2', 'console_scripts', 'kolibri')()
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 673, in main
    start(port, daemon=daemon)
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 281, in start
    update()
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 260, in update
    update_channel_metadata()
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/annotation.py", line 48, in update_channel_metadata
    import_channel_from_local_db(channel_id)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/channel_import.py", line 673, in import_channel_from_local_db
    import_manager = initialize_import_manager(channel_id, cancel_check=cancel_check)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/channel_import.py", line 647, in initialize_import_manager
    channel_metadata = read_channel_metadata_from_db_file(get_content_database_file_path(channel_id))
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/channels.py", line 59, in read_channel_metadata_from_db_file
    source = Bridge(sqlite_file_path=channeldbpath)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/sqlalchemybridge.py", line 216, in __init__
    db_matches_schema(self.Base, self.session)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/check_schema_db.py", line 24, in db_matches_schema
    tables = iengine.get_table_names()
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/reflection.py", line 201, in get_table_names
    self.bind, schema, info_cache=self.info_cache)
  File "<string>", line 2, in get_table_names
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/reflection.py", line 54, in cache
    ret = fn(self, con, *args, **kw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/dialects/sqlite/base.py", line 1160, in get_table_names
    rs = connection.execute(s)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 2075, in execute
    return connection.execute(statement, *multiparams, **params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 942, in execute
    return self._execute_text(object, multiparams, params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 1104, in _execute_text
    statement, parameters
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 1200, in _execute_context
    context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
    exc_info
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/util/compat.py", line 265, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/util/compat.py", line 248, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 1193, in _execute_context
    context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/default.py", line 509, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.DatabaseError: (sqlite3.DatabaseError) database disk image is malformed [SQL: "SELECT name FROM sqlite_master WHERE type='table' ORDER BY name"] (Background on this error at: http://sqlalche.me/e/4xp6)
```

**Comment by rtibbles** [comment]
Fixed in #5419
